test/tests/functional/pbs_cray_vnode_per_numa.py

```diff
@@ -0,0 +1,169 @@
+                dncpus[n['resources_available.host']
+                       ] += int(n['resources_available.ncpus'])
+            if n['resources_available.host'] not in dmem.keys():
+                dmem[n['resources_available.host']] = int(
+                    n['resources_available.mem'][0:-2])
+            else:
+                dmem[n['resources_available.host']
+                     ] += int(n['resources_available.mem'][0:-2])
+            if 'resources_available.naccelerators' in n.keys():
+                if n['resources_available.naccelerators'][0] != '@':
+                    if n['resources_available.host'] not in dacc.keys():
+                        dacc[n['resources_available.host']] = int(
+                            n['resources_available.naccelerators'])
+                    else:
+                        dacc[n['resources_available.host']
+                             ] += int(n['resources_available.naccelerators'])
+
+        # Remove the configuration setting and re-read the vnodes
+        rv = self.mom.unset_mom_config('$vnode_per_numa_node', False)
+        self.assertTrue(rv)
+        self.reset_nodes(momname)
+
+        vnodes_combined = self.server.status(NODE)
+
+        # Compare the multiple vnodes values to the combined vnode output
+
+        for n in vnodes_combined:
+            if 'resources_available.PBScrayseg' in n:
+                self.logger.info(
+                    "ERROR resources_available.PBScrayseg was found.")
+                self.assertTrue(False)
+
+            self.assertEqual(int(n['resources_available.ncpus']), dncpus[
+                n['resources_available.host']])
+            self.assertEqual(int(n['resources_available.mem'][0:-2]), dmem[
+                n['resources_available.host']])
+            if 'resources_available.naccelerators' in n:
+                self.assertEqual(int(n['resources_available.naccelerators']),
+                                 dacc[n['resources_available.host']])
+
+    def restartPBS(self):
+        try:
+            svcs = PBSInitServices()
+            svcs.restart()
+        except PbsInitServicesError, e:
+            self.logger.info("PBS restart failed: \n" + e.msg)
+            self.assertTrue(e.rv)
+
+    def get_momname(self):
+        """
+        Get name of mom host.
+        """
+        momA = self.moms.values()[0]
+        hostA = momA.shortname
+        self.logger.info('mom is: %s ', hostA)
+        return hostA
+
+    def reset_nodes(self, hostA):
+        """
+        Reset nodes.
+        """
+
+        # Remove all nodes
+        rv = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rv, 0)
+
+        # Restart PBS
+        self.restartPBS()
+
+        # Create node
+        rv = self.server.manager(MGR_CMD_CREATE, NODE, None, hostA)
+        self.assertEqual(rv, 0)
+        # cmd = [qmgr, '-c', "create node %s" % (hostA)]
+
+        # Wait for 3 seconds for changes to take effect
+        time.sleep(3)
```

Comment:
The nodes take some time to come up, this allows PBS to finish creating the information on the nodes.